key.js

```diff
@@ -1,106 +1,350 @@
+// GAME STATE INFO :
+// 0. Assign this IIFE's return value to a variable
 
-const buttonPress = () => {
-    const audio = new Audio("GameAudio/buttonPress.wav");
-    audio.currentTime = 0;
-    audio.play();
-}
 
-const setComplete = () => {
-    const audio = new Audio("GameAudio/setComplete.wav");
-    audio.currentTime = 0;
-    audio.play();
-}
-const invalidInput = () => {
-    const audio = new Audio('GameAudio/invalid.flac')
-    audio.currentTime = 0;
-    audio.play();
-}
-const clickSoundGreen = () => {
-    const audio = new Audio('GameAudio/greenSound2.mp3');
-    audio.currentTime = 0;
-    audio.play();
-}
-const clickSoundRed = () => {
-    const audio = new Audio('GameAudio/MPOP.wav')
-    audio.currentTime = 0;
-    audio.play();
-}
-// KeyPress Event :
-
-addEventListener('keydown', (e) => {
-    const direction = e.key;
-    if (direction[5] == 'U') {
-        dUp.style.transform = `scale(.95) translateY(-3px)`;
-        dUp.style.boxShadow = `0px 0px 10px 5px black inset`;
-        dUp.style.transition = `.3s ease all`;
+const gameModule = (function () {
+    const getInitialGameState = () => ({
+        lives: 3,
+        seconds: 60,
+        levelsInfo: {
+            currentLevel: 0,
+            totalLevels: 5,
+            setInfoList: [1, 2, 3, 6, 8, 11]
+        },
+        isStart: false,
+        isGameOver: false,
+        sound: true,
+        gameAudio: {
+            getGameAudio: () => new Audio("GameAudio/gameBackgroundMusic.mp3"),
+            getClickAudio: () => new Audio("GameAudio/buttonPress.wav"),
+            getStartSound: () => new Audio("GameAudio/start_sound.wav"),
+            getKeyboardSounds: {
+                getInvalidAudio: () => new Audio('GameAudio/invalid.flac'),
+                getGreenAudio: () => new Audio()(`GameAudio/greenSound2.mp3`),
+                getRedAudio: () => new Audio("GameAudio/MPOP.wav")
+            }
+        },
+    });
+    let gameState = getInitialGameState();
+    function startGame() {
+        gameState.isStart = true;
+        gameState.currentLevel = 0;
+        gameState.gameAudio.getClickAudio();
+    }
+    function gameOver() {
+        if ((!gameState.lives || !gameState.seconds) && !gameState.levelsInfo.currentLevel) {
+            resetGameState();
+        }
+        if ((!gameState.lives || !gameState.seconds) && gameState.levelsInfo.currentLevel > 0) {
+            resetGameLives();
+            resetGameTimer();
+        }
+    }
+    function levelUp() {
+        // Level 0 -> 5:
+        const currentLevel = gameState.levelsInfo.currentLevel;
+        if (gameState.lives > 0 && gameState.seconds > 0) {
+            resetGameLives();
+            resetGameTimer();
+            gameState.levelsInfo.currentLevel++;
+            increaseSetCount();
+        }
+    }
+    function increaseSetCount() {
+        let level = gameState.levelsInfo.currentLevel;
+        gameState.levelsInfo.setInfoList[level];
+    }
```

Comment:
Either return this value, or create a new state variable in `gameState` that represents the `currentState` and do the following:

```suggestion
    function increaseSetCount() {
        gameState.levelsInfo.currentSet = gameState.levelsInfo.setsInEachLevel[gameState.levelsInfo.currentLevel];
    }
```